src/Views/Form/Themes/Sequoia/assets/js/form.js

```diff
@@ -1,39 +1,175 @@
+			} else {
+				$advanceButton.show();
+			}
 
-			// Show remain form options.
-			$( 'form > *:not(.give-section.choose-amount)', $container ).show();
+			steps[ step ].setup();
+			navigator.currentStep = step;
+		},
+		back: () => {
+			const prevStep = navigator.currentStep !== 0 ? navigator.currentStep - 1 : 0;
+			navigator.goToStep( prevStep );
+			navigator.currentStep = prevStep;
+		},
+		forward: () => {
+			const nextStep = navigator.currentStep !== null ? navigator.currentStep + 1 : 1;
+			navigator.goToStep( nextStep );
+			navigator.currentStep = nextStep;
+		},
+	};
 
-			$parent.removeClass( 'give-showing_choose-amount-section' ).addClass( 'give-showing__personal-section' );
-		}
+	const steps = [
+		{
+			id: 'introduction',
+			title: 'Introduction',
+			selector: '.give-section.introduction, .give-section.income-stats, .give-section.progress-bar',
+			label: templateOptions.introduction.donate_label,
+			setup: () => {
+
+			},
+		},
+		{
+			id: 'choose-amount',
+			title: 'Choose Amount',
+			selector: '.give-section.choose-amount',
+			label: templateOptions.payment_amount.next_label,
+			setup: () => {
+				$( '.give-donation-level-btn' ).each( function() {
+					const hasTooltip = $( this ).attr( 'has-tooltip' );
+					if ( hasTooltip ) {
+						return;
+					}
+
+					const value = $( this ).attr( 'value' );
+					const text = $( this ).text();
+					if ( value !== 'custom' ) {
+						const wrap = `<span class="give-tooltip hint--top hint--bounce" style="width: 100%" aria-label="${ text }" rel="tooltip"></span>`;
+						const html = `<div class="currency">$</div>${ value }`;
+						$( this ).html( html );
+						$( this ).wrap( wrap );
+						$( this ).attr( 'has-tooltip', true );
+					}
+				} );
+				$( '.give-total-wrap', $container ).addClass( 'give-flex' );
+				$( '.give-donation-levels-wrap', $container ).addClass( 'give-grid' );
+			},
+		},
+		{
+			id: 'personal',
+			title: 'Add Your Information',
+			label: 'Process Donation',
+			selector: '.give-section.personal, #give_checkout_user_info, #give-payment-mode-select, #give_purchase_form_wrap',
+			setup: () => {
+				// Show remain form options.
+				$( '.give-label' ).html( '' );
+				$( 'label[for=give-first]' ).html( '<i class="fas fa-user"></i>' );
+				$( 'label[for=give-email]' ).html( '<i class="fas fa-envelope"></i>' );
+			},
+		},
+	];
 
+	const styles = {
+		setup: () => {
+			// Setup custom styles stylesheet
+			const sheet = ( function() {
+				// Create the <style> tag
+				const style = document.createElement( 'style' );
+
+				// WebKit hack :(
+				style.appendChild( document.createTextNode( '' ) );
+
+				// Add the <style> element to the page
+				document.head.appendChild( style );
+
+				return style.sheet;
+			}() );
+
+			const primaryColor = templateOptions.introduction.primary_color ? templateOptions.introduction.primary_color : '#28C77B';
+
+			// Insert rules to custom stylesheet
+			sheet.insertRule( `.seperator {
```

Comment:
Can we run `insertRule` function only once instead of multiple?

Reply:
To my knowledge insertRule only supports adding one rule at a time. See spec: https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule